File: app/src/main/java/com/anadolstudio/adelaide/view/screens/dialogs/ImageDialog.java
package com.anadolstudio.adelaide.view.screens.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.graphics.Bitmap;
import android.graphics.Point;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatDialogFragment;

import com.anadolstudio.adelaide.R;
import com.anadolstudio.adelaide.databinding.DialogImageBinding;
import com.anadolstudio.adelaide.domain.utils.BitmapUtil;

public class ImageDialog extends AppCompatDialogFragment {
    public static final String TAG = ImageDialog.class.getName();
    private static final String PHOTO_FILE = "photo_file";
    private DialogImageBinding binding;
    private String path;

    public static ImageDialog newInstance(String path) {
        Bundle args = new Bundle();
        args.putString(PHOTO_FILE, path);
        ImageDialog fragment = new ImageDialog();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        binding = DialogImageBinding.inflate(getLayoutInflater());
        path = getArguments().getString(PHOTO_FILE);
        Point size = new Point();
        getActivity().getWindowManager().getDefaultDisplay().getSize(size);
        int min = Math.min(size.x, size.y);

        Bitmap bitmap = BitmapUtil.decodeBitmapFromPath(getActivity(), path, min, min);
        binding.imagePhoto.getLayoutParams().height = bitmap.getHeight();
        binding.imagePhoto.requestLayout();

        binding.imagePhoto.setImageBitmap(bitmap);

        AlertDialog dialog = new AlertDialog.Builder(getActivity())
                .setView(binding.getRoot())
                .create();

        if (dialog.getWindow() != null)
            dialog.getWindow().getAttributes().windowAnimations = R.style.SlidingDialogAnimation;

        return dialog;
    }
}

File: app/src/main/java/com/anadolstudio/adelaide/view/screens/dialogs/ImageDialog.kt
package com.anadolstudio.adelaide.view.screens.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.graphics.Point
import android.os.Bundle
import androidx.appcompat.app.AppCompatDialogFragment
import com.anadolstudio.adelaide.R
import com.anadolstudio.adelaide.databinding.DialogImageBinding
import com.anadolstudio.core.util.BitmapDecoder
import kotlin.math.min

class ImageDialog : AppCompatDialogFragment() {

    companion object {

        private const val PHOTO_FILE = "photo_file"

        fun newInstance(path: String?): ImageDialog = ImageDialog().apply {
            val args = Bundle()
            args.putString(PHOTO_FILE, path)
            arguments = args
        }

    }

    private lateinit var binding: DialogImageBinding
    private var path: String? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        binding = DialogImageBinding.inflate(layoutInflater)

        path = arguments?.getString(PHOTO_FILE) ?: throw IllegalArgumentException()

        val size = Point()
        requireActivity().windowManager.defaultDisplay.getSize(size)
        val min = min(size.x, size.y)

        val bitmap = BitmapDecoder.Manager.decodeBitmapFromPath(
            requireContext(),
            path!!, min, min
        )

        binding.imagePhoto.layoutParams.height = bitmap.height
        binding.imagePhoto.requestLayout()
        binding.imagePhoto.setImageBitmap(bitmap)

        val dialog: AlertDialog = AlertDialog.Builder(activity)
            .setView(binding.root)
            .create()

        if (dialog.window != null)
            dialog.window!!.attributes.windowAnimations = R.style.SlidingDialogAnimation

        return dialog
    }

}